internal/collector/postgres.go

```diff
@@ -84,14 +84,25 @@ func EnablePostgresLogging(
 		// PostgreSQL v8.3 adds support for CSV logging, and
 		// PostgreSQL v15 adds support for JSON logging. The latter is preferred
 		// because newlines are escaped as "\n", U+005C + U+006E.
-		if inCluster.Spec.PostgresVersion < 15 {
+		if inCluster != nil && inCluster.Spec.PostgresVersion < 15 {
 			outParameters.Mandatory.Add("log_destination", "csvlog")
 		} else {
 			outParameters.Mandatory.Add("log_destination", "jsonlog")
 		}
 
-		// Keep seven days of logs named for the day of the week;
-		// this has been the default produced by `initdb` for some time now.
+		var retentionPeriod *v1beta1.Duration
+		if inCluster != nil && inCluster.Spec.Instrumentation != nil &&
+			inCluster.Spec.Instrumentation.Logs != nil &&
+			inCluster.Spec.Instrumentation.Logs.RetentionPeriod != nil {
+			retentionPeriod = inCluster.Spec.Instrumentation.Logs.RetentionPeriod
+		} else {
+			retentionPeriod, _ = v1beta1.NewDuration("1d")
```

Comment:
I wish I could create a `Duration` struct literal so that I wouldn't need to handle an error... @cbandy is this possible?

Reply:
🤔 I recommend changing `collector.parseDurationForLogrotate` to accept `metav1.Duration` or `time.Duration`.

Reply:
Done. Mind taking a look?